spec/commands/add_spec.rb

```diff
@@ -79,4 +79,28 @@
       expect(the_bundle).to include_gems "foo 2.0"
     end
   end
+
+  it "using combination of short form options works like long form" do
+    bundle "add 'foo' -s='file://#{gem_repo2}' -g='development' -v='~>1.0'"
+    expect(bundled_app("Gemfile").read).to match(%r{gem "foo", "~> 1.0", :group => \[:development\], :source => 'file:\/\/#{gem_repo2}'})
+    expect(the_bundle).to include_gems "foo 1.1"
+  end
+
+  it "shows error message when version is not formatted correctly" do
+    bundle "add 'foo' -v='~>1 . 0'"
+    expect(out).to match('Illformed requirement ["~>1 . 0"]')
```

Comment:
Change to a user friendly error... something like "Invalid version specified. Please enter a valid version."